rfcs/20251103-level-zero-runtime/README.md

```diff
@@ -0,0 +1,184 @@
+# Support for Level Zero GPU runtime (RFC)
+
+## Introduction
+
+This RFC discusses the option of adding the Intel® oneAPI Level Zero (Level Zero) as fully supported oneDNN GPU runtime.
+
+Level Zero is a fully featured API that provides direct-to-metal interfaces to offload Intel® accelerator devices (similarly to OpenCL™). As such, it is the preferred API for providing explicit device controls for runtime APIs and libraries using Intel® GPUs. For example, Level Zero is the default backend of SYCL runtime.
+
+For oneDNN, Level Zero would provide quicker access to the newest features available for Intel® accelerators, including: reusable and mutable command lists, support for multi-device contexts, shared memory between accelerators, as well as lowered execution latency.
+
+Using a modified [oneDNN example](https://github.com/uxlfoundation/oneDNN/blob/main/examples/cnn_inference_f32.cpp) and [POC implementation](https://github.com/uxlfoundation/oneDNN/tree/spalicki/l0_backend), the Level Zero runtime shows better performance than OpenCL™ and SYCL:
+
+|           | Level Zero | OpenCL™ | SYCL with Level Zero | SYCL with OpenCL™ |
+| --------- | ---------- | ------- | -------------------- | ----------------- |
+| Min (ms): |        557 |     640 |                  599 |               658 |
+| Avg (ms): |        579 |     668 |                  621 |               689 |
+
+## Proposal
+
+### Build option
+
+Adding Level Zero runtime does not require new build options and only requires a new runtime `L0` in the existing CMake build option `ONEDNN_GPU_RUNTIME`, and it and assumes that: `ONEDNN_GPU_VENDOR=INTEL`.
+
+### API/ABI change
+
+Adding Level Zero as a new oneDNN runtime is fairly invisible to the end user and requires no API/ABI modification.
+
+### Integration with SYCL runtime
+
+Currently, the Level Zero loader is used when oneDNN is compiled with SYCL runtime and is using the Level Zero backend (similarly to OpenCL™ runtime when using the OpenCL™ backend), since not all required features are available directly in the SYCL API. The common code of SYCL with Level Zero and native Level Zero would be extracted as part of the new Level Zero runtime (to avoid redundancies) in the same way common OpenCL™ code is part of the OpenCL™ runtime.
+
+### Integration with OpenCL™ runtime
+To use OpenCL™ C kernels, there are several common functions that need to be extracted from oneDNN OpenCL™ runtime code and moved to `src/gpu/intel/compute/utils.hpp`:
+```cpp
+// Defined in src/gpu/intel/ocl/engine.hpp implemented in src/gpu/intel/ocl/engine.cpp
+status_t preprocess_headers(stringstream_t &pp_code, const char *code, const compute::kernel_ctx_t &kernel_ctx);
+
+// Defined and implemented in src/gpu/intel/ocl/utils.cpp
+void debugdump_processed_source(const std::string &source, const std::string &options, const std::string &cl_options);
+```
+
+### Interop API
```

Comment:
Do we also need to expose `cache_blob` related functionalities for engine serialization?